tests/test_batch.py

```diff
@@ -131,6 +139,21 @@ def test_with_no_blocks(mock_task, users):
     assert mock_task.call_count == 1
 
 
+@patch('completion_aggregator.tasks.aggregation_tasks.update_aggregators.apply_async')
+def test_lock(mock_task, users):
```

Comment:
Good call adding a test for this. Can you add one for the cleanup as well?

Reply:
Sure, I'll add this :+1: